include/wlr/types/wlr_surface_layers.h

```diff
@@ -0,0 +1,74 @@
+#ifndef _WLR_SURFACE_LAYERS_H
+#define _WLR_SURFACE_LAYERS_H
+
+#include <wayland-server.h>
+
+struct wlr_surface_layers {
+	struct wl_global *wl_global;
+	struct wl_list surfaces; // list of wlr_layer_surface
+
+	struct {
+		struct wl_signal new_surface;
+	} events;
+
+	void *data;
+};
+
+enum wlr_layer_surface_layer {
+	WLR_LAYER_SURFACE_LAYER_BACKGROUND,
+	WLR_LAYER_SURFACE_LAYER_BOTTOM,
+	WLR_LAYER_SURFACE_LAYER_TOP,
+	WLR_LAYER_SURFACE_LAYER_OVERLAY,
+};
+
+enum wlr_layer_surface_input_device {
+	WLR_LAYER_SURFACE_INPUT_DEVICE_NONE = 0,
+	WLR_LAYER_SURFACE_INPUT_DEVICE_POINTER = 1,
+	WLR_LAYER_SURFACE_INPUT_DEVICE_KEYBOARD = 2,
+	WLR_LAYER_SURFACE_INPUT_DEVICE_TOUCH = 3,
+};
+
+enum wlr_layer_surface_anchor {
+	WLR_LAYER_SURFACE_ANCHOR_NONE = 0,
+	WLR_LAYER_SURFACE_ANCHOR_TOP = 1,
+	WLR_LAYER_SURFACE_ANCHOR_BOTTOM = 2,
+	WLR_LAYER_SURFACE_ANCHOR_LEFT = 4,
+	WLR_LAYER_SURFACE_ANCHOR_RIGHT = 8,
+};
```

Comment:
I should probably just include `surface_layers_server_protocol.h` instead.

Reply:
Yes

Reply:
Hum, I have trouble doing so because it's a header file not installed on the host system but generated on build. Any idea how to solve this?

Reply:
You should get it via https://github.com/swaywm/wlroots/blob/master/types/meson.build#L26

Reply:
Okay, so I shouldn't include it in the header file. I wanted to avoid this because you loose the `enum wlr_layer_surface_layer layer` field (you have to replace that by `uint32_t layer`).